### ISSUE TITLE
Support multiple transporters

### ISSUE THREAD

**Issue by gendronb** [lependu/fastify-nodemailer#17]
Hi!

Any project in supporting multiple transporters (eg. Sparkpost, nodemailer-sparkpost-transport)?

By default, fastify.nodemailer could be the default nodemail transport, but additional transporters could be accessible via fastify.nodemailer.sometransporter, or fastify.nodemailer.transporters(sometransporter)?

Thanks!

**Comment by lependu** [comment]
Hi! Thanks for checking this out. In theory you can setup any custom transporter, just pass it to the plugin instead of an option object:

```js
'use strict'

const fastify = require('fastify')()
const nodemailer = require('fastify-nodemailer')
const stubTransport = require('nodemailer-stub-transport')

// Register the plugin and pass the stub transporter
fastify.register(nodemailer, stubTransport())
```

I do not test it extensively though, so please let me know if it is working with your setup. Also you did not mention your use case, if you need further help, please give me a simple example to start with.

**Comment by gendronb** [comment]
@lependu 

Great to hear! I will try this early this week, please stay tuned.

As for multiple transporters, I have a project in which, depending on the route, I need to email using either the default transporter or the Sparkpost transporter, so being able to register multiple transports and access them on will would be great...

Also, on a multi-tenant project, each tenant could setup its own transporter (with different API key and endpoints in the case of Sparkpost), so again, being able to register multiple transporters could be great. Or maybe simpler, being able to access the nodemailer instance to create new transporterd dynamically?

**Comment by gendronb** [comment]
@lependu 

Hi again...

This works perfectly when registering a custom transporter as mentionned (tested with nodemailer-sparkport-transporter). Thanks!

The possibility of registering multiple transporters would still be useful...

My 2 cents...

**Comment by lependu** [comment]
Sorry for the long response times i am a bit occupied. Good to hear that it really works. :tada: 

You can register multiple transporters using Fastify built in [encapsulation](https://www.fastify.io/docs/latest/Plugins-Guide/#register) feature. All you need to do is register this plugin with  your custom transporter and the corresponding route in new context.

On the other hand i think we should document that the plugin works with custom transporters in the `README.md` and also add a test case for it. Would you like to make a PR?

**Comment by gendronb** [comment]
Agree...

I will update the docs (including your mention regarding encapsulation) and send you a PR.

As for the test case, I have one already for the Sparkpost transport, but including it would leak my API key...

**Comment by lependu** [comment]
Thank you, i will review it as fast as it possible. From 25 of march I will be offline for one week though.

I think a minimal transporter mock would be enough for test. Something like [this](https://nodemailer.com/plugins/create/#transport-example). In that way we don't need any additional dependency or expose api key.

**Comment by lependu** [comment]
@gendronb  Thank you for the PR's, they are merged in #18 #19 and #20.
If you like the idea, please add yourself to project maintainers in package.json.
Closing this for now, feel free to reopen if necessary.
